sql/2020/07_SEO/lighthouse.sql

```diff
@@ -8,6 +8,27 @@ CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS
   ROUND(SAFE_DIVIDE(freq, total), 4)
 );
 
+CREATE TEMPORARY FUNCTION isCrawlable(report STRING) 
+RETURNS STRUCT<
+disallow BOOL, 
+noindex BOOL,
+both BOOL,
+neither BOOL
+> DETERMINISTIC LANGUAGE js AS '''
+var result = {disallow:false, noindex:false};
+try {
+var $ = JSON.parse(report);
+var items = $.audits['is-crawlable'].details.items;
+result.noindex = items.filter(item => item.source.type ==='node').length > 0;
+result.disallow = items.filter(item => item.source.type ==='source-location').length > 0;
+result.both = result.noindex && result.disallow;
+result.neither = !result.noindex && !result.disallow;
+} catch (e) {
```

Comment:
Let's indent for readability:

```suggestion
try {
    var $ = JSON.parse(report);
    var items = $.audits['is-crawlable'].details.items;
    result.noindex = items.filter(item => item.source.type ==='node').length > 0;
    result.disallow = items.filter(item => item.source.type ==='source-location').length > 0;
    result.both = result.noindex && result.disallow;
    result.neither = !result.noindex && !result.disallow;
} catch (e) {
```